src/Microsoft.IdentityModel.Tokens/BaseConfigurationManager.cs

```diff
@@ -76,9 +93,39 @@ public BaseConfiguration LastKnownGoodConfiguration
             {
                 _lastKnownGoodConfiguration = value ?? throw LogHelper.LogArgumentNullException(nameof(value));
                 _lastKnownGoodConfigFirstUse = DateTime.UtcNow;
+
+                if (_lkgConfigurationCache == null)
+                    _lkgConfigurationCache = new Dictionary<BaseConfiguration, DateTime>(BaseConfigurationComparer);
+                
+                _lkgConfigurationCache[_lastKnownGoodConfiguration] = DateTime.UtcNow + LastKnownGoodLifetime;
```

Comment:
>_lkgConfigurationCache [](http://example.com/codeflow?start=16&length=22)

Is there any sane upper bound we can add on this cache? Currently it's only bounded by the LastKnownGoodLifetime correct? This goes back to the complexity functionality tradeoff but seems reasonable to have a high upper bound here which can be overridden to prevent runaway cache in some error case where ESTS starts doing something weird with metadata publishing.

Reply:
We could set the upper limit for the cache of lavage our EventBaseLruCache.